braidpool-primitives/src/utils/test_utils.rs

```diff
@@ -0,0 +1,204 @@
+#[cfg(test)]
+use super::BeadHash;
+#[cfg(test)]
+use crate::bead::TimeVec;
+#[cfg(test)]
+use crate::bead::{Bead, CommittedMetadata, UnCommittedMetadata};
+#[cfg(test)]
+use bitcoin::BlockHeader;
+#[cfg(test)]
+use bitcoin::ecdsa::Signature;
+#[cfg(test)]
+use bitcoin::p2p::Address as P2P_Address;
+#[cfg(test)]
+use bitcoin::{PublicKey, Transaction, absolute::Time, p2p::address::AddrV2};
+#[cfg(test)]
+pub mod test_utility_functions {
+    use std::collections::HashSet;
+
+    use bitcoin::CompactTarget;
+
+    use super::*;
+    pub struct TestUnCommittedMetadataBuilder {
+        extra_nonce: i32,
+        broadcast_timestamp: Option<Time>,
+        signature: Option<Signature>,
+        parent_bead_timestamps: Option<TimeVec>,
+    }
+
+    #[cfg(test)]
+    impl TestUnCommittedMetadataBuilder {
+        pub fn new() -> Self {
+            Self {
+                extra_nonce: 0,
+                broadcast_timestamp: None,
+                signature: None,
+                parent_bead_timestamps: None,
+            }
+        }
+
+        pub fn extra_nonce(mut self, nonce: i32) -> Self {
```

Comment:
Why are there all these getter/setter functions? It's very verbose and useless code.

Can we do this without a bunch of getter/setters?